doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -395,6 +565,62 @@ stands for :n:`let @ident := fun {+ @binder} => @term in @term’`.
 Definition by case analysis
 ---------------------------
 
+.. insertgram match_constr pattern_list_or2
+
+.. productionlist:: coq
+   match_constr : match `case_item_list_comma` `term100_opt` with `or_opt` `eqn_list_or_opt` end
+   case_item_list_comma : `case_item_list_comma` , `case_item`
+                        : `case_item`
+   term100_opt : `term100`
+               : `empty`
+   case_item : `term100` `as_opt` `in_opt`
+   as_opt : as `name`
+          : `empty`
+   in_opt : in `pattern`
+          : `empty`
+   or_opt : |
+          : `empty`
+   eqn_list_or_opt : `eqn_list_or`
+                   : `empty`
+   eqn_list_or : `eqn_list_or` | `eqn`
+               : `eqn`
+   eqn : `pattern_list_or` => `term`
+   pattern_list_or : `pattern_list_or` | `pattern_list_comma`
+                   : `pattern_list_comma`
+   pattern_list_comma : `pattern_list_comma` , `pattern`
+                      : `pattern`
+   record_pattern : `qualid` := `pattern`
+   record_patterns : `record_pattern` ; `record_patterns`
+                   : `record_pattern` ;
+                   : `record_pattern`
+                   : `empty`
+   pattern : `pattern10` : `binder_constr`
+           : `pattern10` : `term100`
```

Comment:
You missed a possible simplification here: these two rules could have been conflated into ``` `pattern10` : `term` ```

Reply:
Will do.  But worth keeping in mind that if someone adds a new production to `operconstr200`, that may invalidate this and the similar simplification we discussed in your earlier comments.

FWIW, I would not have used "conflate" here, I probably would have said "combined".  In my experience, "conflate" is generally used when referring to two things being combined that don't really fit together.  In https://www.merriam-webster.com/dictionary/conflate, for example,  meaning 1b is "confuse".  And see the part "Recent Examples on the Web".  Looking at the first several links, only about half mention confuse as a meaning, though.

Reply:
Now the simplification has been merged in the parser itself.

Reply:
Yup